source/usage-examples/findOne.rst

```diff
@@ -0,0 +1,43 @@
+===============
+Find a Document
+===============
+
+.. default-domain:: mongodb
+
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+You can find a single document using the ``collection.findOne()``
+method. If no document is found, ``findOne`` returns ``null``. If a
+document is found, ``findOne`` returns that document as an Object.
+The following snippet finds a single document from the ``movies``
+collection:
+
+.. code-block:: javascript
+
+   const { MongoClient } = require('mongodb');
```

Comment:
Ran this code through eslint with the airbnb config and got the following errors:
```
   1:33  error    Unable to resolve path to module 'mongodb'                       import/no-unresolved
   4:1   error    Expected indentation of 2 spaces but found 3                     indent
   4:79  error    Missing trailing comma                                           comma-dangle
   8:1   error    Expected indentation of 2 spaces but found 3                     indent
   9:1   error    Expected indentation of 4 spaces but found 6                     indent
  10:1   error    Expected indentation of 4 spaces but found 6                     indent
  10:34  error    Strings must use singlequote                                     quotes
  11:1   error    Expected indentation of 4 spaces but found 6                     indent
  11:7   error    All 'var' declarations must be at the top of the function scope  vars-on-top
  11:7   error    Unexpected var, use let or const instead                         no-var
  11:44  error    Strings must use singlequote                                     quotes
  12:1   error    Expected indentation of 4 spaces but found 6                     indent
  12:46  error    A space is required after '{'                                    object-curly-spacing
  12:47  error    Extra space after key 'title'                                    key-spacing
  12:47  error    Unnecessarily quoted property 'title' found                      quote-props
  12:47  error    Strings must use singlequote                                     quotes
  12:57  error    Strings must use singlequote                                     quotes
  12:67  error    A space is required before '}'                                   object-curly-spacing
  13:1   error    Expected indentation of 4 spaces but found 6                     indent
  14:1   error    Expected indentation of 4 spaces but found 6                     indent
  14:7   warning  Unexpected console statement                                     no-console
  15:1   error    Expected indentation of 2 spaces but found 3                     indent
  16:1   error    Expected indentation of 4 spaces but found 6                     indent
  17:1   error    Expected indentation of 2 spaces but found 3                     indent
  19:13  warning  Unexpected console statement                                     no-console
  19:26  error    Newline required at end of file but not found       
```
Ignoring the console warning and the import warning here is the fixed version of the code:
```
const { MongoClient } = require('mongodb');

const client = new MongoClient(
  'mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority',
);

async function run() {
  try {
    await client.connect();
    const database = client.db('sample_mflix');
    const collection = database.collection('movies');
    const movie = await collection.findOne({ title: 'The Room' });
    // since this method returns the matched document, not a cursor, we can print it directly
    console.log(movie);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);
```

Reply:
Updated. Most of these (indentation, console statement) are results of the fact that this code is nested inside a code block with an extra level of indentation.

PS 2 space tabs are the worst